### PR TITLE
Added a new wake-lock effect

### PR DESCRIPTION
**Pull request by Vynlar**
I added an effect to the `effects.cljs` file for managing a single global wakelock. It supports two actions: `:lock` and `:unlock` which do what you would expect.
I confirmed it works in my browser. I didn't safeguard anything for browsers that don't support the `wakeLock` API but this is a good start.

@kolyaventuri

### CODE REVIEW

File: src/cljs/coffee_re_frame/pages/recipe_setup.cljs [hunk 1]
```diff
@@ -18,61 +18,59 @@
...
-                 [c/home-button]]
-                [c/micro-header {:for "volume" :as :label} "How much coffee do you want to make?"]
-                [:div {:class "h-16"}
-                 (if (= :custom @volume-type)
-                   [:input {:id "custom-volume"
-                            :class "text-3xl font-bold bg-gray-700 border-none rounded px-2 py-1 w-full"
-                            :type :number
-                            :value @volume
-                            :on-change #(re-frame/dispatch [:recipe-setup/set-volume (parse-number-event %)])}]
+         [:form {:class "flex flex-col p-4 space-y-2"}
```
Comment (Vynlar) [inline src/cljs/coffee_re_frame/pages/recipe_setup.cljs:21]: Sorry this is all auto-formatting 😭

**Review by kolyaventuri — APPROVED** [review]
This is much cleaner than what I had goin'! Thanks :) Looks good to me

File: src/cljs/coffee_re_frame/effects.cljs [hunk 2]
```diff
@@ -37,3 +37,31 @@
   (.vibrate js/navigator duration))
 
 (re-frame/reg-fx :vibrate vibrate-effect)
+
+(def current-lock (atom nil))
+
+(defn lock-brightness []
+  (->
+   (.request (.-wakeLock js/navigator) "screen")
```
Comment (kolyaventuri) [inline src/cljs/coffee_re_frame/effects.cljs:45]: This we could wrap in an `(if (exists? js/navigator.wakeLock))` clause

File: src/cljs/coffee_re_frame/effects.cljs [hunk 3]
```diff
@@ -37,3 +37,31 @@
...
+   (.then (fn [new-lock]
+            (js/console.log "Locking screen brightness: " new-lock)
+            (reset! current-lock new-lock)))
+   (.catch (fn [error]
+             (js/console.error "Error locking screen brightness"
+                               error)))))
```
Comment (kolyaventuri) [inline src/cljs/coffee_re_frame/effects.cljs:51]: Useful for dev work, but honestly if we don't have reporting, we could just throw this into a black hole

**Comment by Vynlar** [comment]
Did some refactoring:
- Moved the call to `:wakelock` into the `engine` instead of it being inside the button handler on the setup page. That way the lock gets engaged even if you go directly to the brew URL.
- Release the wake lock when we hit a step with `:step/type` equal to `:step.type/end` (which every recipe should have)

Take a peek -- I think this is good to merge

File: src/cljs/coffee_re_frame/effects.cljs [hunk 4]
```diff
@@ -37,3 +37,33 @@
...
 (re-frame/reg-fx :vibrate vibrate-effect)
+
+(def current-lock (atom nil))
+
+(defn lock-brightness []
+  (when-let [wakeLock (.-wakeLock js/navigator)]
+    (->
+     (.request wakeLock "screen")
+     (.then (fn [new-lock]
+              (js/console.log "Locking screen brightness: " new-lock)
```
Comment (kolyaventuri) [inline src/cljs/coffee_re_frame/effects.cljs:48]: [nit]: Do we not want these to be conditional debug logs? Makes little sense to log them in prod

**Review by kolyaventuri — APPROVED** [review]
LGTM

**Pull request merged**